src/macros.rs

```diff
@@ -1,8 +1,11 @@
 /// Variable argument version of `syscall`
 #[macro_export]
 macro_rules! syscall {
-    ($nr:ident, $a1:expr, $a2:expr) => {
-        $crate::syscall($crate::nr::$nr, &($a1 as usize))
+    ($nr:ident) => {
+        $crate::syscall($crate::nr::$nr, 0usize)
```

Comment:
I suppose that for syscall with zero arguments you don't need to touch R1 at all?